simulations/vip-diamond-comptroller-testnet/simulations.ts

```diff
@@ -0,0 +1,461 @@
+      });
+
+      it("Diamond Unitroller Implementation (comptroller) should match the diamond Proxy Address", async () => {
+        const comptrollerImplementation = await unitroller.comptrollerImplementation();
+        console.log("implementation address", comptrollerImplementation);
+        const pendingComptrollerImplementation = await unitroller.pendingComptrollerImplementation();
+        expect(comptrollerImplementation.toLowerCase()).to.equal(
+          "0xc934A1b15b30E9b515D8A87b5054432B9b965131".toLowerCase(),
+        );
+        expect(pendingComptrollerImplementation.toLowerCase()).to.equal(zeroAddr);
+      });
+    });
+
+    describe("Verify storage layout before vip execution", async () => {
+      it("verify all the state before and after upgrade", async () => {
+        oracle = await unitroller.oracle();
+
+        maxAssets = await unitroller.maxAssets();
+
+        closeFactorMantissa = await unitroller.closeFactorMantissa();
+
+        liquidationIncentiveMantissa = await unitroller.liquidationIncentiveMantissa();
+
+        allMarkets = await unitroller.allMarkets(0);
+
+        venusRate = await unitroller.venusRate();
+
+        venusSpeeds = await unitroller.venusSpeeds(BUSD.address);
+
+        venusSupplyState = await unitroller.venusSupplyState(BUSD.address);
+
+        venusBorrowState = await unitroller.venusBorrowState(BUSD.address);
+
+        venusAccrued = await unitroller.venusAccrued(BUSD.address);
+
+        vaiMintRate = await unitroller.vaiMintRate();
+
+        vaiController = await unitroller.vaiController();
+
+        mintedVAIs = await unitroller.mintedVAIs(busdHolder.address);
+        unitroller.minte;
+
+        mintVAIGuardianPaused = await unitroller.mintVAIGuardianPaused();
+
+        repayVAIGuardianPaused = await unitroller.repayVAIGuardianPaused();
+
+        protocolPaused = await unitroller.protocolPaused();
+
+        venusVAIVaultRate = await unitroller.venusVAIVaultRate();
+
+        vaiVaultAddress = await unitroller.vaiVaultAddress();
+
+        releaseStartBlock = await unitroller.releaseStartBlock();
+
+        minReleaseAmount = await unitroller.minReleaseAmount();
+
+        treasuryGuardian = await unitroller.treasuryGuardian();
+
+        treasuryAddress = await unitroller.treasuryAddress();
+
+        treasuryPercent = await unitroller.treasuryPercent();
+
+        liquidatorContract = await unitroller.liquidatorContract();
+
+        comptrollerLens = await unitroller.comptrollerLens();
+
+        // cheking all public mappings
+        market = await unitroller.markets(vBUSD.address);
+
+        venusBorrowerIndex = await unitroller.venusBorrowerIndex(vBUSD.address, busdHolder.address);
+
+        venusSupplierIndex = await unitroller.venusSupplierIndex(vBUSD.address, busdHolder.address);
+
+        venusBorrowSpeeds = await unitroller.venusBorrowSpeeds(vUSDT.address);
+        venusSupplySpeeds = await unitroller.venusSupplySpeeds(vUSDT.address);
```

Comment:
I would do this for every market returned by `unitroller.getAllMarkets`, not only for vBUSD and vUSDT

Reply:
Added for all markets, cfe4633bf5ca7d1e1d013d6b9dcd7206e3050a02